frontend/src/metabase-lib/lib/queries/NativeQuery.js

```diff
@@ -45,11 +45,11 @@ export const NATIVE_QUERY_TEMPLATE: NativeDatasetQuery = {
 };
 
 // This regex needs to match logic in replaceCardId and _getUpdatedTemplateTags.
-const CARD_TAG_REGEX = /^#\s*([0-9]*)\s*$/;
+const CARD_TAG_REGEX = /^#([0-9]*)$/;
```

Comment:
We should still allow whitespace before and after `#([0-9]*)`, no? It's allowed in other template tag types, and the two other regexes below.

Reply:
Yes space around `#[0-9]*` is still allowed. This regex is used after that group is extracted from [here](https://github.com/metabase/metabase/blob/11804-reference-question-as-cte/frontend/src/metabase-lib/lib/queries/NativeQuery.js#L326).

Reply:
Just to confirm: was the `*` this regex deliberately chosen? I.e. shouldn't we require at least one number for a match (`+`)?

Reply:
We parse `{{#}}` as a card tag, but if the captured group is `""`, we set the card_id to null. That's triggers the "Pick a question" UI.